### ISSUE TITLE
qwen2-audio 微调 Bug

### ISSUE THREAD

**Issue by ucas-hao** [teamtee/Qwen2-Audio-finetune#3]
您好，您代码中：
text_ids = processor(prompt,return_tensors="pt", padding=True)
是否应该是
text_ids = processor(prompt,audios=audio,return_tensors="pt", padding=True)

同时
for i,attention_mask  in enumerate(text_ids["attention_mask"]):
        labels[i,:sum(attention_mask)] = -100
这一步是否有问题
比如我的labels[0] 是['151643','151643','151643','151647','151646'*16,'151648','1','2','3',...]
这时候我的sum(attention_mask)对应的长度应该是['151647','151646'*16,'151648','1','2','3',...]中instruction部分的长度，
接着在使用labels[i,:sum(attention_mask)] = -100会错误的将'151643','151643','151643'这些长度的-100给漏掉

实际上应该是sum(attention_mask)+labels[0]中'151643'的个数作为-100

**Comment by teamtee** [comment]
Could you describe the problem you encountered in more detail? I haven't noticed the problem myself.
<img width="959" height="199" alt="Image" src="https://github.com/user-attachments/assets/9cd625a1-7b98-4d7c-b920-e7fa71579551" />

**Comment by ucas-hao** [comment]
src/dataset.py 62行
text_ids = processor(prompt,return_tensors="pt", padding=True)
应该是text_ids = processor(prompt,audios=audio,return_tensors="pt", padding=True)

我写的collate_fn_qwen2audio

def collate_fn_qwen2audio(samples,processor):
    prompt = [_["prompt"] for _ in samples] # list of '<|audio_bos|><|AUDIO|><|audio_eos|>Detect the language and recognize the speech: <|zh|>'
    audio = [ _["audio"] for _ in samples] # list of array
    target = [ _["target"] for _ in samples] # ['楼市调控的行政手段宜减不宜加', '以及提供服务中的变相涨价或价格欺诈行为', '积极防范和妥善应对市场价格异常波动', '在加快体制改革方面']
    processed_data = processor(text=[i+j for i,j in zip(prompt,target)], audios=audio, sampling_rate=processor.feature_extractor.sampling_rate, return_tensors="pt", padding=True)
    # processed_data: input_ids attention_mask input_features feature_attention_mask,这里是pad到max
    processed_attention_mask = processed_data["attention_mask"]
    labels = copy.deepcopy(processed_data["input_ids"])
    text_ids = processor(prompt,audios=audio,return_tensors="pt", padding=True) # torch.Size([4, 155])
    for i,attention_mask  in enumerate(text_ids["attention_mask"]):
        len1 = (processed_attention_mask[i] == 0).sum().item()
        len2 = sum(attention_mask).item()
        labels[i,:(len1 + len2)] = -100
    processed_data["labels"]=labels
    return  processed_data

欢迎大佬交流

**Comment by teamtee** [comment]
I did make a mistake here, but your solution still has issues. I will describe it to you in detail later. I'm sorry, but as a matter of fact, I haven't used this library for a long time, and my work has shifted to other methods.

First, the length in text_ids will not be affected whether audio is added or not (you can try it when the batch_size is 1):
```
text_ids = processor(prompt, return_tensors="pt", padding=True)
text_ids = processor(prompt, audios=audio, return_tensors="pt", padding=True)
```
Second, the token you saw: 151643 is actually the pad token. It appears because there are prompts + targets of different lengths, so there will be padding on the left.

Therefore, the solution is to add the length of the padding to labels[i, :sum(attention_mask)] = -100:
```
    processed_data = processor(text=[i + j for i, j in zip(prompt, target)], audios=audio, sampling_rate=processor.feature_extractor.sampling_rate, return_tensors="pt", padding=True)
    labels = copy.deepcopy(processed_data["input_ids"])
    text_ids = processor(prompt, return_tensors="pt", padding=True)
    for i, attention_mask in enumerate(text_ids["attention_mask"]):
        labels[i, :len(attention_mask)] = -100
```

**Comment by teamtee** [comment]
I will fix it today.

**Comment by teamtee** [comment]
Done

**Comment by zhhao1** [comment]
> I did make a mistake here, but your solution still has issues. I will describe it to you in detail later. I'm sorry, but as a matter of fact, I haven't used this library for a long time, and my work has shifted to other methods.
> 
> First, the length in text_ids will not be affected whether audio is added or not (you can try it when the batch_size is 1):
> 
> ```
> text_ids = processor(prompt, return_tensors="pt", padding=True)
> text_ids = processor(prompt, audios=audio, return_tensors="pt", padding=True)
> ```
> 
> Second, the token you saw: 151643 is actually the pad token. It appears because there are prompts + targets of different lengths, so there will be padding on the left.
> 
> Therefore, the solution is to add the length of the padding to labels[i, :sum(attention_mask)] = -100:
> 
> ```
>     processed_data = processor(text=[i + j for i, j in zip(prompt, target)], audios=audio, sampling_rate=processor.feature_extractor.sampling_rate, return_tensors="pt", padding=True)
>     labels = copy.deepcopy(processed_data["input_ids"])
>     text_ids = processor(prompt, return_tensors="pt", padding=True)
>     for i, attention_mask in enumerate(text_ids["attention_mask"]):
>         labels[i, :len(attention_mask)] = -100
> ```

First, the length in text_ids will not be affected whether audio is added or not (you can try it when the batch_size is 1):

这句话有问题。
processor(prompt, audios=audio, return_tensors="pt", padding=True)这行代码传参数传错了，应该是audio=audio。
processing_qwen2_audio.py的代码中没有audios这个参数，所以传audios识别不了，代码就判断没有处理音频，导致text_ids不变。

**Comment by teamtee** [comment]
You can refer to the [example](https://github.com/QwenLM/Qwen2-Audio) given by Qwen2-audio official, their code is audios.

<img width="1144" height="605" alt="Image" src="https://github.com/user-attachments/assets/361d0cbf-f41e-4d47-a40f-e298122bcdf7" />

**Comment by zhhao1** [comment]
> You can refer to the [example](https://github.com/QwenLM/Qwen2-Audio) given by Qwen2-audio official, their code is audios.
> 
> <img alt="Image" width="1144" height="605" src="https://private-user-images.githubusercontent.com/87838510/491995737-361d0cbf-f41e-4d47-a40f-e298122bcdf7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg0MzE4NzYsIm5iZiI6MTc1ODQzMTU3NiwicGF0aCI6Ii84NzgzODUxMC80OTE5OTU3MzctMzYxZDBjYmYtZjQxZS00ZDQ3LWE0MGYtZTI5ODEyMmJjZGY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIxVDA1MTI1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzY2ZkM2MzOWQ1NTFiNTcwMTgwMjdmODViOGVlNzEzZTkwY2VlOTU2ZWEyYWMyOTJkM2JhZDg4Y2ZlNTIyODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9CbBlYPYoh4g7h_POCRA6GtzgtiVrWl6PsAsOzj_Rko">

可能版本问题。。现在的版本要传audio，你可以自己调试一下，

<img width="1057" height="190" alt="Image" src="https://github.com/user-attachments/assets/702ab2b7-40e2-4817-8986-a877de40febb" />
得到的inputs反解码出来这个样子才是正常的，

> Could you describe the problem you encountered in more detail? I haven't noticed the problem myself. <img alt="Image" width="959" height="199" src="https://private-user-images.githubusercontent.com/87838510/468453334-9cd625a1-7b98-4d7c-b920-e7fa71579551.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg0MzE4NzYsIm5iZiI6MTc1ODQzMTU3NiwicGF0aCI6Ii84NzgzODUxMC80Njg0NTMzMzQtOWNkNjI1YTEtN2I5OC00ZDdjLWI5MjAtZTdmYTcxNTc5NTUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTIxVDA1MTI1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwYTgyNWExYTg5ZTQ5YWRkMTU3OGFjMTgyZGY5OWIxODI1ZTFmZmQ5MzJkZjYxMzkxYjk5YTY4ZmMyNjgxNDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Vvatd6AXmrEpbiKgLz1c4YFTs3q7hOJr_WQKvW8DzSI">

这里相当于音频长度是1，是不正确的

**Comment by teamtee** [comment]
This should be related to the version of the Transformer library you installed.
